LEAF_Request_Portal/sources/FormWorkflow.php

```diff
@@ -1193,6 +1197,147 @@ public function handleEvents(int $workflowID, int $stepID, string $actionType, s
+                    $data = $this->buildFileLink($data, $field["indicatorID"], $field["series"]);
+                    break;
+                case ($format == "orgchart_group"):
+                    $data = $this->getOrgchartGroup((int) $data);
+                    break;
+                case ($format == "orgchart_position"):
+                    $data = $this->getOrgchartPosition((int) $data);
+                    break;
+                case ($format == "orgchart_employee"):
+                    $data = $this->getOrgchartEmployee((int) $data);
+                    break;
+            }
+
+            $formattedFields[$field['indicatorID']] = $data !== "" ? $data : $field["default"];
+        }
+
+        return $formattedFields;
+    }
+    
+    // method for building grid
+    private function buildGrid(array $data): string
+    {
+        // get the grid in the form of array
+        $cells = $data['cells'];
+        $headers = $data['names'];
+        
+        // build the grid
+        $grid = "<table><tr>";
+
+        foreach($headers as $header) {
+            if ($header !== " ") {
+                $grid .= "<th>{$header}</th>";
+            }
+        }
+        $grid .= "</tr>";
+
+        foreach($cells as $row) {
+            $grid .= "<tr>";
+            foreach($row as $column) {
+                $grid .= "<td>{$column}</td>";
+            }
+            $grid .= "</tr>";
+        }
+        $grid .= "</table>";
+
+        return $grid;
+    }
+
+    private function buildMultiselect(array $data): string
+    {
+        // filter out non-selected selections
+        $data = array_filter($data, function($x) { return $x !== "no"; });
+        // comma separate to be readable in email
+        $formattedData = implode(",", $data);
+
+        return $formattedData;
+    }
+
+    private function buildFileLink(string $data, string $id, string $series): string
+    {
+        // split the file names out into an array
+        $data = explode("\n", $data);
+        $buffer = [];
+
+        // parse together the links to each file
+        foreach($data as $index => $file) {
+            $buffer[] = "<a href=\"{$this->siteRoot}file.php?form={$this->recordID}&id={$id}&series={$series}&file={$index}\">{$file}</a>";
+        }
+
+        // separate the links by comma
+        $formattedData = implode(", ", $buffer);
+        return $formattedData;
+    }
+
+    // method for building orgchart group, position, employee
+    private function getOrgchartGroup(int $data): string
+    {
+        // reference the group by id
+        $group = new Group($this->db, $this->login);
+        $groupName = $group->getGroupName($data);
+        
+        return $groupName;
+    }
+
+    private function getOrgchartPosition(int $data): string
+    {
+        $position = new \Orgchart\Position($this->oc_db, $this->login);
+        $positionName = $position->getTitle($data);
+
+        return $positionName;
+    }
+

```

Comment:
I would suggest we make sure we have phpdocs here as well. Helps with adding context when needed and helps fill in the intellisense a bit more with what is actually going on.